tests/cxx/array.cc

```diff
@@ -24,8 +24,27 @@ class A {
     // IWYU: IndirectClass is...*indirect.h
     return &(_b[i]);
   }
+
+  // IWYU: IndirectTemplate needs a declaration
+  // IWYU: IndirectClass needs a declaration
+  IndirectTemplate<IndirectClass> *getIndirectTemplateSpecialization(int i) {
+    // IWYU: IndirectTemplate is...*indirect.h
+    // IWYU: IndirectClass is...*indirect.h
+    (void)sizeof(_t[i]);  // requires full type
+    // IWYU: IndirectTemplate needs a declaration
+    // IWYU: IndirectTemplate is...*indirect.h
+    // IWYU: IndirectClass is...*indirect.h
+    (void)sizeof(&(_t[i]));  // requires full type
+    // IWYU: IndirectTemplate is...*indirect.h
+    // IWYU: IndirectClass is...*indirect.h
+    return &(_t[i]);
+  }
+
   // IWYU: IndirectClass needs a declaration
   IndirectClass *_b;
+  // IWYU: IndirectTemplate needs a declaration
+  // IWYU: IndirectClass needs a declaration
+  IndirectTemplate<IndirectClass> *_t;
```

Comment:
`_t` is a reserved name, per https://en.cppreference.com/w/cpp/language/identifiers ("[...] identifiers that begin with an underscore are reserved in the global namespace"), so let's pick a better name. Even `t` would work in this context.

Side note: `_b` has the same problem, and if you want to fix that up in a separate commit, that'd be great.